docs/style-guide.rst

```diff
@@ -0,0 +1,223 @@
+
+    * Use terms consistently.
+    * Avoid ambiguous pronouns.
+    * Eliminate unneeded words.
+    * Establish key points at the start of a document.
+    * Focus each paragraph on a single topic.
+    * Focus each sentence on a single idea.
+    * Use a numbered list when order is important and a bulleted list when order is irrelevant.
+    * Introduce lists and tables appropriately.
+
+Google's `technical writing courses <https://developers.google.com/tech-writing>`_ are a valuable resource. We recommend reviewing them before any significant documentation work.
+
+API Directives
+--------------
+
+    * All API documentation **must** use standard Python `directives <https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-python-domain>`_.
+    * Where possible, references to syntax **should** use appropriate `Python roles <https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax>`_.
+    * External references **may** use `intersphinx roles <https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html>`_.
+
+Headers
+-------
+
+    * Each documentation section **must** begin with a `label <https://www.sphinx-doc.org/en/stable/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations>`_ of the same name as the filename for that section. For example, this section's filename is ``style-guide.rst``, so the RST opens with a label ``_style-guide``.
+    * Section headers **should** use the following sequence, from top to bottom: ``#``, ``=``, ``-``, ``*``, ``^``.
+
+Internal Documentation
+======================
+
+Internal documentation is vital to aid other contributors in understanding the layout of the Vyper codebase.
+
+We handle internal documentation in the following ways:
+
+    * A ``README.md`` **must** be included in each first-level subdirectory of the Vyper package. The readme explain the purpose, organization and control flow of the subdirectory.
+    * All publicly exposed classes and methods **must** include detailed docstrings.
+    * Internal methods **should** include docstrings, or at minimum comments.
+    * Any code that may be considered "clever" or "magic" **must** include comments explaining exactly what is happening.
+
+Docstrings **should** be formatted according to the `NumPy docstring style <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+Commit Messages
+===============
+
+Contributors **should** adhere to the following standards and best practices when making commits to be merged into the Vyper codebase.
+
+Maintainers  **may** request a rebase, or choose to squash merge pull requests that do not follow these standards.
+
+Conventional Commits
+--------------------
+
+Commit messages **should** adhere to the `Conventional Commits <https://www.conventionalcommits.org/en/v1.0.0-beta.4/>`_ standard. A convetional commit message is structured as follows:
```

Comment:
Looks like v1.0 is actually released, so I've given you the wrong link :grimacing:
```suggestion
Commit messages **should** adhere to the `Conventional Commits <https://www.conventionalcommits.org/>`_ standard. A convetional commit message is structured as follows:
```